tests/unit/utils/test_serialize_dataclass.py

```diff
@@ -0,0 +1,217 @@
+
+        :returns: The function used for conversion to a dictionary or nothing
+        """
+        return self._asdict(suffix=f"_y_{ContentView.FULL!s}")
+
+    def serialize_yaml_normal(self) -> OverrideDictReturn:
+        """Provide dictionary factory for ``JSON`` with curated attributes.
+
+        :returns: The function used for conversion to a dictionary or nothing
+        """
+        return self._asdict(suffix=f"_y_{ContentView.NORMAL!s}")
+
+    def _asdict(self, suffix) -> OverrideDictReturn:
+        return asdict(self, dict_factory=partial(self._custom_dict_factory, suffix=suffix))
+
+    @staticmethod
+    def _custom_dict_factory(
+        kv_pairs: List[Tuple[str, OverrideAllValuesT]],
+        suffix: str,
+    ) -> OverrideDictReturn:
+        """Create a dictionary with suffixed values from a list of key-value pairs.
+
+        :param kv_pairs: The key-value pairs provided by ``dataclasses.asdict()``
+        :param suffix: The suffix to append to values
+        :returns: The dictionary with suffixed values
+        """
+        return {name: f"{str(value)}_{suffix}" for name, value in kv_pairs}
+
+
+@pytest.mark.parametrize(**ParametrizeFormat()._asdict())
+@pytest.mark.parametrize(**ParametrizeView()._asdict())
+def test_content_to_dict(
+    content_view: ContentView,
+    serialization_tuple: Tuple[str, SerializationFormat],
+):
+    """Test the conversion of the dataclass to a dict.
+
+    :param content_view: The content view
+    :param serialization_tuple: The suffix, serialization format
+    """
+    content = ContentTestSimple()
+    result = content.asdict(
+        serialization_format=serialization_tuple[1],
+        content_view=content_view,
+    )
+    assert result == {"attr_01": False, "attr_02": 2, "attr_03": "three"}
+
+
+@pytest.mark.parametrize(**ParametrizeView()._asdict())
+def test_content_to_json(content_view: ContentView):
+    """Test the conversion of the dataclass to json.
+
+    :param content_view: The content view
+    """
+    content = ContentTestSimple()
+    result = serialize(
+        content=content,
+        content_view=content_view,
+        serialization_format=SerializationFormat.JSON,
+    )
+    expected = '{\n    "attr_01": false,\n    "attr_02": 2,\n    "attr_03": "three"\n}'
+    assert result == expected
+
+
+@pytest.mark.parametrize(**ParametrizeView()._asdict())
+def test_content_to_yaml(content_view: ContentView):
+    """Test the conversion of the dataclass to yaml.
+
+    :param content_view: The content view
+    """
+    content = ContentTestSimple()
+    result = serialize(
+        content=content,
+        content_view=content_view,
+        serialization_format=SerializationFormat.YAML,
+    )
+    expected = "---\nattr_01: false\nattr_02: 2\nattr_03: three\n"
+    assert result == expected
+
+
+@pytest.mark.parametrize(**ParametrizeFormat()._asdict())
+@pytest.mark.parametrize(**ParametrizeView()._asdict())
```

Comment:
I see this is repeated a few times so for such cases, consider assigning the mark to a variable and decorate with that:
```python
with_view_params = pytest.mark.parametrize(**ParametrizeView()._asdict())  # <-- put this somewhere above the test declarations
```
```suggestion
@with_view_params
```

Reply:
Wow.... Hadn't considered that... I'll do this.

Reply:
So much cleaner :)